File: ValidParentheses.java
public class Solution {
    public boolean isValid(String s) {
        if (s.length()% 2 != 0) return false;
        Stack stack = new Stack();
        for (int i = 0 ; i < s.length(); i ++) {
            if(stack.empty() || s.charAt(i) == '(' || s.charAt(i) == '{' || s.charAt(i)=='['){
                stack.push(s.charAt(i));
            }
            if(s.charAt(i) == ')' && (char)stack.pop() != '(') return false;
            if(s.charAt(i) == '}' && (char)stack.pop() != '{') return false;
            if(s.charAt(i) == ']' && (char)stack.pop() != '[') return false;
        }
        if (stack.empty()) return true;
        return false;
    }
}

File: ValidParentheses.py
class Solution(object):
    def isValid(self, s):
        stack = []
        for c in s:
            if c in ['(','[','{']:
                stack.push(c)
            else:
                if not stack:
                    return False
                temp = stack.pop()
                if temp == '(' and c != ')':
                    return False
                if temp == '[' and c!= ']':
                    return False
                if temp == '{' and c != '}':
                    return False
        if len(stack) > 0:
            return False
        else:
            return True